homeassistant/auth/__init__.py

```diff
@@ -481,6 +502,8 @@ async def async_validate_access_token(
         if refresh_token is None or not refresh_token.user.is_active:
             return None
 
+        self.async_validate_refresh_token(refresh_token, remote)
+
```

Comment:
This is somewhat debatable. In general, access tokens don't normally go all the way up to the refresh token when they validate. They should normally be self sustaining to reduce load on the authentication server. But the code already looks up the refresh token here, so it should be quite easy to re-validate here.

That said. If we do some more complicated auth provider that needs to do some heavy lifting to validate, this might be a bad idea? (would also remove the need to pass in the remote)

Reply:
Seems by default our access tokens live for 30 minutes. So i think it should be safe to drop this check, only check on renewal.

Reply:
If we should check here. we should encode whatever is needed to do a validation in the access token itself, and just ask provider to validate the access token.